src/components/EventModal/Collect/index.vue

```diff
@@ -564,6 +636,17 @@ export default {
         }
       }
     },
+    handleAddToCart() {
+      logTrackerEvent(
+        this,
+        'NFT',
+        'nft_collect_modal_click_add_to_cart',
+        this.classId,
+        1
+      );
+      this.addNFTClassToShoppingCart({ classId: this.classId });
+      this.isAddedToShoppingCart = true;
```

Comment:
Need to set `this.paymentMethod` before triggering tx, otherwise there will be a close button that allow user to close the dialog too early

Reply:
Updated